sdk/storage/azblob/zt_blob_versioning_test.go

```diff
@@ -112,70 +110,71 @@ func TestCreateAndDownloadBlobSpecialCharactersWithVID(t *testing.T) {
-//				snapTime = *blob.Snapshot
-//			}
-//			fmt.Printf("Blob name: %s, Snapshot: %s\n", *blob.Name, snapTime)
-//		}
-//	}
-//	_assert.Nil(listPager.Err())
-//	_assert.Len(count, 2)
-//
-//	// Deleting previous version snapshot.
-//	deleteResp, err := blobURL.WithVersionID(versionID1).Delete(ctx, DeleteSnapshotsOptionNone, BlobAccessConditions{})
-//	_assert.NoError(err)
-//	_assert(deleteResp.StatusCode(), chk.Equals, 202)
-//
-//	listBlobsResp, err = containerClient.ListBlobsFlat(ctx, Marker{}, ListBlobsSegmentOptions{Details: BlobListingDetails{Versions: true}})
-//	_assert.NoError(err)
-//	_assert(listBlobsResp.Segment.BlobItems, chk.NotNil)
-//	if len(listBlobsResp.Segment.BlobItems) != 1 {
-//		s.T().Fail()
-//	}
-//}
-//
+func TestDeleteSpecificBlobVersion(t *testing.T) {
+	stop := start(t)
+	defer stop()
+
+	testName := t.Name()
+	svcClient, err := createServiceClient(t, testAccountDefault)
+	require.NoError(t, err)
+
+	containerName := generateContainerName(testName)
+	containerClient := createNewContainer(t, containerName, svcClient)
+	defer deleteContainer(t, containerClient)
+	bbClient := getBlockBlobClient(generateBlobName(testName), containerClient)
+
+	versions := make([]string, 0)
+	for i := 0; i < 5; i++ {
+		uploadResp, err := bbClient.Upload(ctx, internal.NopCloser(bytes.NewReader([]byte("data"+strconv.Itoa(i)))), &UploadBlockBlobOptions{
+			Metadata: basicMetadata,
+		})
+		require.NoError(t, err)
+		require.NotNil(t, uploadResp.VersionID)
+		versions = append(versions, *uploadResp.VersionID)
+	}
+
+	listPager := containerClient.ListBlobsFlat(&ContainerListBlobFlatSegmentOptions{
+		Include: []ListBlobsIncludeItem{ListBlobsIncludeItemVersions},
+	})
+
+	found := make([]*BlobItemInternal, 0)
+	for listPager.NextPage(ctx) {
+		resp := listPager.PageResponse()
+		found = append(found, resp.ContainerListBlobFlatSegmentResult.Segment.BlobItems...)
+	}
+	require.NoError(t, listPager.Err())
+	require.Len(t, found, 5)
+
+	// Deleting the 2nd and 3rd versions
+	for i := 0; i < 3; i++ {
+		deleteResp, err := bbClient.WithVersionID(versions[i]).Delete(ctx, nil)
+		require.NoError(t, err)
+		require.Equal(t, deleteResp.RawResponse.StatusCode, 202)
+	}
+
+	listPager = containerClient.ListBlobsFlat(&ContainerListBlobFlatSegmentOptions{
+		Include: []ListBlobsIncludeItem{ListBlobsIncludeItemVersions},
+	})
+
+	found = make([]*BlobItemInternal, 0)
+	for listPager.NextPage(ctx) {
+		resp := listPager.PageResponse()
+		found = append(found, resp.ContainerListBlobFlatSegmentResult.Segment.BlobItems...)
+	}
+	require.NoError(t, listPager.Err())
+	require.Len(t, found, 2)
+
+	for i := 3; i < 5; i++ {
+		downloadResp, err := bbClient.WithVersionID(versions[i]).Download(ctx, nil)
+		require.NoError(t, err)
+		destData, err := ioutil.ReadAll(downloadResp.Body(nil))
+		require.NoError(t, err)
+		require.EqualValues(t, destData, "data"+strconv.Itoa(i))
+	}
+}
+
 //func (s *azblobTestSuite) TestDeleteSpecificBlobVersionWithBlobSAS() {
```

Comment:
Can you uncomment these tests as well? What is the blocker on having them commented out?